### PR TITLE
Prep Prefect for Python 3.8 Compatibility

### PR DESCRIPTION
**Pull request by joshmeek**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
Fixing tests and and issues which may arise from testing Prefect on Python 3.8

## Why is this PR important?
Python 3.8 support is right around the corner!

### CODE REVIEW

**Comment by joshmeek** [comment]
Unit tests on 3.8 look to be passing. Need to update the circle steps to include this in the coverage and do some more QA in terms of runs w/ and w/o Cloud but otherwise looks smooth so far 👍

**Comment by wagoodman** [comment]
heads up: I saw an issue running specifically python 3.8 with the docker agent (an import error related to `pypiwin32`). A quick test would be to start the docker agent and confirm that there are no import errors.

**Comment by cicdw** [comment]
This is great - we should add a docker build step for a 3.8 image too.

**Comment by joshmeek** [comment]
Ran a bunch of checks to see if things still operated as planned. Tested quite a few prefect core related functionality (schedules, skips, serialization, etc.) to make sure (even though they were never expected to break). 

Also tested some Dask runs w/ local, static k8s, dask-k8s. The only issue I encountered was a case where my flow was stored into a Python 3.8 image and my static Dask cluster was running Python 3.7
```
Unexpected error: ValueError('unsupported pickle protocol: 5')
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/prefect/engine/runner.py", line 48, in inner
    new_state = method(self, state, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/prefect/engine/flow_runner.py", line 462, in get_flow_run_state
    final_states = executor.wait(
  File "/usr/local/lib/python3.8/site-packages/prefect/engine/executors/dask.py", line 196, in wait
    return self.client.gather(futures)
  File "/usr/local/lib/python3.8/site-packages/distributed/client.py", line 1887, in gather
    return self.sync(
  File "/usr/local/lib/python3.8/site-packages/distributed/client.py", line 779, in sync
    return sync(
  File "/usr/local/lib/python3.8/site-packages/distributed/utils.py", line 348, in sync
    raise exc.with_traceback(tb)
  File "/usr/local/lib/python3.8/site-packages/distributed/utils.py", line 332, in f
    result[0] = yield future
  File "/usr/local/lib/python3.8/site-packages/tornado/gen.py", line 735, in run
    value = future.result()
  File "/usr/local/lib/python3.8/site-packages/distributed/client.py", line 1752, in _gather
    raise exception.with_traceback(traceback)
  File "/usr/local/lib/python3.7/site-packages/distributed/protocol/pickle.py", line 59, in loads
ValueError: unsupported pickle protocol: 5
```
This _may be_ expected behavior when running something across different Python versions but thought I would note here in case it is not. After updating the Dask cluster to Python 3.8 all worked as planned.

Other than that no hiccups along the way 👍

**Comment by cicdw** [comment]
@joshmeek yea dask will even error out if different versions of _dask_ are running, so seeing this error with different versions of python is unsurprising.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,10 +11,11 @@ These changes are available in the [master branch](https://github.com/PrefectHQ/
 ### Enhancements
 
 - Add volume option to Docker Agent - [#2013](https://github.com/PrefectHQ/prefect/issues/2013)
+- Add support for Python 3.8 - [#2080](https://github.com/PrefectHQ/prefect/pull/2080)
```
Comment (cicdw) [inline CHANGELOG.md]: I think this deserves to be a feature

**Pull request merged**